Index: lib/loader.c
===================================================================
--- lib/loader.c	(revision 16538)
+++ lib/loader.c	(working copy)
@@ -259,10 +259,15 @@
 						&& !GLEW_OES_compressed_ETC1_RGB8_texture) {
 						GLubyte* unpacked;
 
-						_ktxUnpackETC(data, &unpacked, pixelWidth, pixelHeight);
+						errorCode = _ktxUnpackETC(data, &unpacked, pixelWidth, pixelHeight);
+						if (errorCode != KTX_SUCCESS) {
+							goto cleanup;
+						}
 						glTexImage2D(texinfo.glTarget + face, level, 
 									 GL_RGB, pixelWidth, pixelHeight, 0, 
 									 GL_RGB, GL_UNSIGNED_BYTE, unpacked);
+
+						free(unpacked);
 					} else
 #endif
 						glCompressedTexImage2D(texinfo.glTarget + face, level, 
